tests/unit/domain/test_message_operations_service_part2.py

```
@@ -0,0 +1,304 @@
+        filtered = [
+            m
+            for m in message_collection.messages.values()
+            if (now - m.timestamp).total_seconds() <= timeout
+        ]
+        return message_collection.create(filtered)
+
+
+class FakeLimitsService:
+    def apply_limits(self, messages, limits):
+        if limits.max_messages:
+            messages = messages[-limits.max_messages :]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)